climpred/classes.py

```diff
@@ -152,6 +153,113 @@ def __repr__(self):
         else:
             return _display_metadata(self)
 
+    def _math(self, other, operator):
+        """Helper function for __add__, __sub__, __mul__, __truediv__.
+
+        Allows math operations with type:
+            - int
+            - float
+            - np.ndarray
+            - xr.DataArray without new dimensions
+            - xr.Dataset without new dimensions or variables
+
+        """
+        assert isinstance(operator, str)
+
+        def add(a, b):
+            return a + b
+
+        def sub(a, b):
+            return a - b
+
+        def mul(a, b):
+            return a * b
+
+        def div(a, b):
+            return a / b
+
+        ALLOWED_TYPES_FOR_MATH_OPERATORS = [
+            int,
+            float,
+            np.ndarray,
+            xr.DataArray,
+            xr.Dataset,
+            type(self),
+        ]
+        # catch other dimensions in other
+        error_str = f'Cannot use {type(self)} {operator} {type(other)}'
+        if isinstance(other, tuple([xr.Dataset, xr.DataArray])):
+            all_dims = set(
+                [d for d in list(self._datasets['initialized'].dims)] + CLIMPRED_DIMS
```

Comment:
What is the reasoning behind this? Shouldn't the `other` doing the operation just have dims that are a subset of the main dataset dims? Such that it can be broadcasted? I would think that `xarray` will just catch if the dims are funky.